src/code/reducers/drakes.js

```diff
@@ -24,16 +24,15 @@ export default function drakes(state = initialState, action) {
           secondXAlleles = GeneticsUtils.computeExtraAlleles(oldOrg, newOrg);
           alleleString = newOrg.getAlleleString();
         }
-        else if (drakeDef.sex === BioLogica.MALE && action.newSex === BioLogica.FEMALE) {
+        else if (drakeDef.sex === BioLogica.MALE && action.newSex === BioLogica.FEMALE && drakeDef.secondXAlleles) {
```

Comment:
That's a nice catch Kirk- you're right that there should be a test for this if I'm going to say here that an empty `secondXAlleles` is valid. It definitely would have caught this issue.

I've never had the chance to try out TDD- it would be nice to do that for my next project.